File: src/test/java/at/cpickl/gadsu/testinfra/TestViewStarter.java
package at.cpickl.gadsu.testinfra;

import java.awt.Component;

import at.cpickl.gadsu.view.components.Framed;
import org.testng.Assert;

public class TestViewStarter {

    public static Component componentToShow;

    // invoked by ui tests
    public static void main(String[] args) {
        if (componentToShow == null) {
            Assert.fail("componentToShow must be set first!");
        }
        Framed.Companion.show(componentToShow, null);
    }

}

File: src/test/kotlin/at/cpickl/gadsu/testinfra/TestViewStarter.kt
package at.cpickl.gadsu.testinfra

import non_test.Framed
import org.testng.Assert
import java.awt.Component

object TestViewStarter {

    var componentToShow: Component? = null

    @JvmStatic
    fun main(args: Array<String>) {
        if (componentToShow == null) {
            Assert.fail("TestViewStarter.componentToShow (static var) must be set first before using it in MainClassAdapter!")
        }
        Framed.show(componentToShow!!, null)
    }
}
